src/ctia/task/migration/store.clj

```diff
@@ -169,24 +178,71 @@
 
 (def bulk-max-size (* 5 1024 1024)) ;; 5Mo
 
+(s/defn store-map->es-conn-state :- ESConnState
+  "Transforms a store map in ES lib conn state"
+  [conn-state :- StoreMap]
+  (dissoc (clojure.set/rename-keys conn-state {:indexname :index})
+          :mapping :type :settings))
+
+
+(defn real-index-of
+  "Retrieves the real index of a document"
+  [{:keys [mapping] :as conn-state} id]
+  (-> (store-map->es-conn-state conn-state)
+      (crud/get-doc-with-index (keyword mapping) id {})
```

Comment:
Doing this for each doc is too heavy. You should group them with a bulk operation.